lib/matplotlib/font_manager.py

```diff
@@ -1349,23 +1349,24 @@ def get_font(filename, hinting_factor=None):
                      _kerning_factor=rcParams['text.kerning_factor'])
 
 
-def _rebuild():
-    global fontManager
-    fontManager = FontManager()
-    with cbook._lock_path(_fmcache):
-        json_dump(fontManager, _fmcache)
+def _load_fontmanager(*, try_read_cache=True):
+    if try_read_cache:
+        fm_path = Path(
+            mpl.get_cachedir(), f"fontlist-v{FontManager.__version__}.json")
+        try:
+            fm = json_load(fm_path)
+        except Exception as exc:
+            pass
+        else:
+            if getattr(fm, "_version", object()) == FontManager.__version__:
+                _log.debug("Using fontManager instance from %s", fm_path)
+                return fm
+    fm = FontManager()
+    with cbook._lock_path(fm_path):
```

Comment:
`fm_path` is only set in the `if`, so this should not work in your new call with `try_read_cache=False`. Additionally, you should keep the lock held for the entirety of the function to prevent race conditions.

Reply:
I'll wait for #16111 to go in first?